src/daemon/ca/keys.rs

```diff
@@ -492,22 +474,28 @@ impl KeyState {
     /// for a newly create pending key and requested certificate for it.
     pub fn keyroll_initiate(
         &self,
-        class_name: ResourceClassName,
+        resource_class_name: ResourceClassName,
         parent_class_name: ResourceClassName,
         base_repo: &RepoInfo,
         name_space: &str,
         signer: &KrillSigner,
-    ) -> KrillResult<Vec<EvtDet>> {
+    ) -> KrillResult<Vec<CaEvtDet>> {
         match self {
             KeyState::Active(_current) => {
-                let key_id = signer.create_key()?;
+                let pending_key = signer.create_key()?;
 
-                let issuance_req =
-                    self.create_issuance_req(base_repo, name_space, parent_class_name, &key_id, signer)?;
+                let req = self.create_issuance_req(base_repo, name_space, parent_class_name, &pending_key, signer)?;
 
                 Ok(vec![
-                    EvtDet::KeyRollPendingKeyAdded(class_name.clone(), key_id),
-                    EvtDet::CertificateRequested(class_name, issuance_req, key_id),
+                    CaEvtDet::KeyRollPendingKeyAdded {
+                        resource_class_name: resource_class_name.clone(),
+                        pending_key,
+                    },
+                    CaEvtDet::CertificateRequested {
+                        resource_class_name,
+                        req,
+                        ki: pending_key,
```

Comment:
Use a longer name for `ki` given that you used the long name `resource_class_name`?